### PR TITLE
fix: Fetch invoices

### LINKED ISSUE #33: Sometimes the connector doesn't get anything
We cannot say why, but it seems that the connector is not well configured to keep authentication along requests.
[trello card](https://trello.com/c/3p7ilnfV/196-leclerc-ne-r%C3%A9cup%C3%A8re-aucun-fichier-dans-la-moiti%C3%A9-des-cas)

### PR DESCRIPTION
**Pull request by enguerran**
This PR fixes #33.

included:

- refactor (remove side-effects)
- dependencies updates

### CODE REVIEW

**Comment by LucsT** [comment]
I see files conflict in dependency, need to be resolve correctly at merge.

File: src/index.js [hunk 1]
```diff
@@ -53,143 +63,169 @@ async function authenticate(login, password) {
...
+    date: dates.date,
+    vendor: 'Leclerc Drive',
+    currency: 'EUR',
+    filename: `${dates.isoDateString}-${String(command.amount).replace(
+      '.',
+      ','
+    )}EUR.pdf`
+  }))
+
+  return Promise.all(
```
[the remark this answers is not in the document]
Comment (enguerran) [reply src/index.js]: I am not sure it is a problem in our context, let's ask @doubleface.

[The behavior was the same before](https://github.com/konnectors/leclercdrive/pull/34/files#diff-1fdf421c05c1140f6d71444ea2b27638L90) I work on the code. I just refactored it to make it more functional.

[the remark this answers is not in the document]
Comment (LucsT) [reply src/index.js]: I mean, to launch request one by one, is more like a browser^^

**Comment by enguerran** [comment]
I don't like renovate. I will work on arguments and experiences to express what I do not like.

**Pull request merged**